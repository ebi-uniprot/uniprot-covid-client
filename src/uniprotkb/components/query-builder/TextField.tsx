import React from 'react';
import { SearchTermType } from '../../types/searchTypes';

type TextFieldProps = {
  field: SearchTermType;
  type: string;
  value?: string;
  handleChange: (value: string) => void;
};

const TextField = ({
  field,
  type,
  handleChange,
  value = '',
}: TextFieldProps) => (
  <div className="advanced-search__inputs" key={field.term}>
    <label htmlFor={`input_${field.term}`}>
      {field.label}
      <input
        id={`input_${field.term}`}
        type={type}
        value={value}
        onChange={e => handleChange(e.target.value)}
        placeholder={field.example}
      />
    </label>
  </div>
);

export default TextField;
