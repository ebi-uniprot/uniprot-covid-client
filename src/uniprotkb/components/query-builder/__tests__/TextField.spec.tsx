import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TextField from '../TextField';

describe('Range field', () => {
  const props = {
    field: {
      label: 'UniProtKB AC',
      itemType: 'single',
      term: 'accession',
      description: 'Search by UniProtKB Accession',
      example: 'P12345',
    },
    type: 'text',
    queryInput: {},
    handleChange: jest.fn(),
  };

  test('should render a text field', () => {
    const updatedValue = 'Some term';
    const { getByPlaceholderText, rerender } = render(<TextField {...props} />);
    const inputElt = getByPlaceholderText('P12345');
    expect(inputElt.value).toBe('');
    fireEvent.change(inputElt, { target: { value: updatedValue } });
    expect(props.handleChange).toBeCalled();
    props.value = updatedValue;
    rerender(<TextField {...props} />);
    expect(inputElt.value).toBe(updatedValue);
  });

  test('should render', () => {
    const { asFragment } = render(<TextField {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
