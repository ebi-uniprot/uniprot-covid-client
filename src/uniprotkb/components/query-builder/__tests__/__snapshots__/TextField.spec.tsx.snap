// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range field should render 1`] = `
<DocumentFragment>
  <div
    class="advanced-search__inputs"
  >
    <label
      for="input_accession"
    >
      UniProtKB AC
      <input
        id="input_accession"
        placeholder="P12345"
        type="text"
        value="Some term"
      />
    </label>
  </div>
</DocumentFragment>
`;
